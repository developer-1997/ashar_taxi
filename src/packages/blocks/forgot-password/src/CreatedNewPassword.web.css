.logoContainerStyle { 
    margin-bottom: 40px;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
.logoImgStyle {
    max-width: 155px;
}
#rightBoxContainer {
    height: 100vh;
    margin: 0;
    padding: 30px;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    box-sizing: border-box;
}
.backToHomeContainer {
    flex-grow: 1;
    width: 100%;
    align-items: flex-end;
}
.backToHomeButton {
    color: #828282;
    font-weight: 600;
    font-size: 18px !important;
    text-decoration: underline !important;
    cursor: pointer;
    padding: 0px !important;
}
.verificationForm {
    padding-left: 60px!important;
    padding-right: 140px!important;
    width: 100% !important;
    box-sizing: border-box;
}
.forgotpassHeader .title {
    width: 100%;
    text-align: center;
}
#newPasswordForm .MuiGrid-spacing-xs-3 > .MuiGrid-item {
    margin-bottom: 12px;
}
@media only screen and (max-width: 900px) {
    #rightBoxContainer {
        height: 50vh;
        padding: 10px 20px;
        align-items: center;
        min-height: 400px;
    }
    .backToHomeButton {
        font-size: 14px !important;
    }
    .logoContainerStyle {  
        margin-bottom: 10px;
    }
    .logoImgStyle {
        max-width: 80px;
    }
    .forgotpassHeader .paragraph {
        font-size: 14px;
    }
    .verificationForm {
        padding-left: 0 !important;
        padding-right: 0 !important;
    }
    #newPasswordForm .MuiGrid-container {
        margin-top: 15px !important;
    }
    .forgotpassHeader {
        margin: 0 !important
    }
    .forgotpassHeader .title {
        width: 100%;
        text-align: center;
        margin-bottom: 10px;
    }
    .forgotpassHeader .title::before {
        height: 3px;
        top: 40px;
    }
    .MuiInputBase-fullWidth {
        height: 50px !important;
        margin-top: 0 !important;
        margin-bottom: 0 !important;
    }
    .MuiGrid-item img {
        top: 12px !important;
    }
    .verificationForm .loginHeader {
        margin-top: 0 !important;
    }
    #newPasswordForm .MuiGrid-spacing-xs-3 > .MuiGrid-item {
        padding-left: 0 !important;
        padding-right: 0 !important;
    }
}

