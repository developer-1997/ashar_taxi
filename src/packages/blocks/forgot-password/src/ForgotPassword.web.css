.logoContainerStyle { 
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
    margin-bottom: 40px;
}
.logoImgStyle {
    max-width: 155px;
}
.contentForm {
    padding-left: 60px;
    padding-right: 140px;
    width: 100%;
    box-sizing: border-box;
}

@media only screen and (max-width: 900px) {
    .logoContainerStyle {  
        margin-top: 0 !important;
        margin-bottom: 20px !important;
    }
    .logoImgStyle {
        max-width: 80px;
    }
    .contentForm {
        padding-left: 0 !important;
        padding-right: 0 !important;
        width: auto !important;
    }
    .forgotpassHeader {
        margin: 0 !important
    }
    .forgotpassHeader .title {
        margin-bottom: 10px;
        font-size: 22px !important;
    }
    .forgotpassHeader .paragraph {
        font-size: 15px;
    }
    .contentForm .MuiGrid-item {
        padding: 12px 0 0 !important;
        box-sizing: border-box;
    }
    .contentForm .MuiInputBase-fullWidth {
        margin: 0 !important;
        height: 48px !important;
        font-size: 14px !important;
    }
    .contentForm img {
        top: 12px !important;
    }
    .contentForm .MuiButton-fullWidth {
        font-size: 16px !important;
        margin-top: 15px !important;
    }
    button.fontWeightMedium {
        right: 0 !important;
    }
    .contentForm .MuiGrid-spacing-xs-3 {
        width: 100% !important;
    }
}
