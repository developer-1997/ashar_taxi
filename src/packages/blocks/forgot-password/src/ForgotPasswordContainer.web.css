#rightBoxContainer {
    height: 100vh;
    margin: 0;
    padding: 30px;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    box-sizing: border-box;
}
.backToHomeContainer {
    flex-grow: 1;
    width: 100%;
    align-items: flex-end;
}
.backToHomeButton {
    color: #828282;
    font-weight: 600;
    font-size: 18px !important;
    text-decoration: underline !important;
    cursor: pointer;
    padding: 0px !important;
}

@media only screen and (max-width: 900px) {
    #rightBoxContainer {
        height: 50vh;
        padding: 10px 20px;
        align-items: center;
        min-height: 400px;
    }
    .backToHomeContainer {
        padding: 20px 0;
    }
    .forgotpassHeader .title {
        margin-bottom: 10px;
    }
    .backToHomeButton {
        font-size: 14px !important;
    }
}
