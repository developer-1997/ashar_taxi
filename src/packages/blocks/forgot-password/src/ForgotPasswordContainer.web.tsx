import React, { Component } from "react";

// Customizable Area Start
import { Box, Button, Grid } from "@material-ui/core";
import { createTheme, ThemeProvider } from "@material-ui/core/styles";
import { Link } from "react-router-dom";
import "./ForgotPasswordContainer.web.css"
import SharableLeftSideTaxi from "../../../components/src/SharableLeftSideTaxi.web";

// Customizable Area End

const theme = createTheme({
  palette: {
    primary: {
      main: "#0000ff",
      contrastText: "#fff"
    }
  }
});

export default class ForgotPassContainer extends Component {
  render() {
    return (
      // Required for all blocks
      <ThemeProvider theme={theme}>
        <Grid container>
          <Grid item md={12} lg={6}>
            <SharableLeftSideTaxi />
          </Grid>
          <Grid item md={12} lg={6}>
            <Box id="rightBoxContainer">
              {this.props.children}
              <Box
                className="backToHomeContainer"
                display="flex"
                flexDirection="row"
                justifyContent="flex-end"
              >
                <Link to="/LandingPage">
                  <Button className="normalText fontWeightMedium backToHomeButton">
                    Back to home
                  </Button>
                </Link>
              </Box>
            </Box>
          </Grid>
        </Grid>
      </ThemeProvider>
    );
  }
}
