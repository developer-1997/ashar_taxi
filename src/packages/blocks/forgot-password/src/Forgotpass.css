.forgotpass .rightBox {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 20px 90px 120px;
    margin: 0px auto;
    max-height: 100%;
    overflow: hidden auto;
    box-sizing: border-box;
}
.forgotpass .rightBox .forgotpassForm {
    max-width: 535px;
    width: 100%;
}
.forgotpassForm .errorConfirmPass label.MuiFormLabel-root, 
.forgotpassForm .errorConfirmPass input, 
.forgotpassForm .errorConfirmPass input::placeholder {
    color: #ff4452 !important;
}
.forgotpass .errorConfirmPass div.MuiInput-root {
    border: 1px solid #ff4452 !important;
}
.forgotpass .brandLogo {
    padding: 40px 30px 55px;
    display: flex;
    align-items: center;
    justify-content: space-between;
}
.forgotpass .brandLogo svg {
    fill:#979797
}
.forgotpass .backbtn {
    position: absolute;
    bottom: 1.875rem;
    right: 6.063rem;
    color: #828282;
    font-weight: 600;
    text-decoration: underline;
    cursor: pointer;
}

.forgotpassHeader {
    margin: 0 0 10px 10px;
}
.forgotpassHeader .title {
    font-size: 24px;
    color: #090168;
    text-transform: capitalize;
    position: relative;
    display: inline-block;
    margin-bottom: 40px;
}
/* .forgotpassHeader .title::before {
    content: "";
    width: 16px;
    height: 6px;
    background-color: #090168;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    top: 45px;
    border-radius: 10px;
} */
.forgotpassHeader .paragraph {
  font-size: 20px;
  color: #8f8f9c;
}
.forgotpassHeader .paragraph b {
    display: block;
    color: #1f1f1f;
}
.forgotpassOtpForm {
    margin-top: 45px !important;
    margin-left: 0px !important;
    margin-right: 0px !important;
}
.forgotpassOtpForm .forgotpassOtpInput {
    display:flex;
}
.forgotpassOtpForm .forgotpassOtpInput > div {
    margin-right: 26px;
}
.forgotpassOtpForm .verifyBtn{
    background-color: #090168;
    padding: 15px;
    border-radius: 10px;
    font-size: 18px;
    text-transform: capitalize;
    margin-top: 20px;
}
.resend {
    color: #090168;
    cursor: pointer;
}
.forgotpassOtpForm .disableBtn{
    background-color: #c8c7c7;
    padding: 15px;    
    border-radius: 10px;
    color:#fff !important;
    font-size: 18px;
    text-transform: capitalize;
    margin-top: 20px;
}
.forgotpassOtpForm .formInputOutline {
    border: 1px solid #c5c5c5;
    padding: 5px 15px;
    border-radius: 5px;
    margin: 5px 0;
    font-size: 30px;
    font-weight: 400;
    max-width: 85px;
    min-height: 78px;
    box-sizing: border-box;
}
.forgotpassOtpForm .IsResend {
    display: flex;
    width: 100%;
    margin-bottom: 40px;
  }
.forgotpassOtpForm .isResendHeading {
    width: 100%;
    font-size: 14px;
    text-align: center;
    font-weight: 700;
    color: #3d3d3d;
  }
@media only screen and (max-width: 1280px) {
    .forgotpass .rightBox .forgotpassForm {
        max-width: 99%;
    }
}
@media only screen and (max-width: 767px) {
    .forgotpass .rightBox {
        padding: 20px 50px 120px 20px;
    }
    .forgotpass .backbtn {
        right: 40px;
    }
}

@media only screen and (max-width: 510px) {
    .forgotpassOtpForm {
        margin-top: 30px auto 0!important;
    }
    .forgotpassOtpForm .verifyBtn{
        margin-top: 10px;
    }
    .forgotpassOtpForm .disableBtn{
        margin-top: 10px;
    }
    .forgotpassOtpForm .forgotpassOtpInput > div {
        margin-right: 12px;
    }
    .forgotpassHeader .paragraph b {
        display: inline;
    }
}
@media only screen and (max-width: 490px) {
    .forgotpass .rightBox {
        padding: 20px 50px 120px 20px;
    }
    .forgotpass .backbtn {
        right: 40px;
    }
}