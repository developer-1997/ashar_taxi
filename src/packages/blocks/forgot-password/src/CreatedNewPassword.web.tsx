import React from "react";

// Customizable Area Start
import { Box } from "@material-ui/core";
import { Formik } from "formik";
import { logoImg } from "../../email-account-registration/src/assets";
import { Link } from "react-router-dom";

import ForgotPasswordController from "./ForgotPasswordController.web";
import ForgotPassContainer from "./ForgotPasswordContainer.web";
import ForgotPassHeader from "./ForgotPasswordHeader.web";
import ArrowBack from "@material-ui/icons/ArrowBack";
import "./CreatedNewPassword.web.css";

// Customizable Area End

export default class CreateNewPassword extends ForgotPasswordController {
  render() {
    return (
      // Required for all blocks
      <ForgotPassContainer>
        <Box
          className="logoContainerStyle"
          display="flex"
          flexDirection="row"
          justifyContent="space-between"
        >
          <Link to="/ForgotPassword">
            <ArrowBack />
          </Link>
          <Link to="/LandingPage">
            <img src={logoImg} alt="Logo" className="logoImgStyle" />
          </Link>
        </Box>
        <Box className="verificationForm">
          <ForgotPassHeader
            title="Reset password"
            paragraph="Please set your new password, making sure it contains at least one uppercase letter, one lowercase letter, one number, and one special character for security"
          />
          <Box id="newPasswordForm" display="flex">
            <Formik
              initialValues={{
                password: "",
                confirmpassword: ""
              }}
              validationSchema={this.newPassvalidation}
              onSubmit={values => {
                this.newPassForForgotPass(
                  values,
                  this.props.location.state.email
                );
              }}
            >
              {this.createdNewPasswordFormHandler}
            </Formik>
          </Box>
        </Box>
      </ForgotPassContainer>
    );
  }
}
