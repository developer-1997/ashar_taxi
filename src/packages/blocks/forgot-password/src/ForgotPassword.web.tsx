import React from "react";

// Customizable Area Start
import {
    Box
} from "@material-ui/core";
import { Formik } from "formik";
import { logoImg } from "../../email-account-registration/src/assets"
import { Link } from "react-router-dom"

import ForgotPasswordController from "./ForgotPasswordController.web";
import ForgotPassContainer from "./ForgotPasswordContainer.web";
import ForgotPassHeader from "./ForgotPasswordHeader.web";
import ArrowBack from '@material-ui/icons/ArrowBack';
import "./ForgotPassword.web.css";

// Customizable Area End
export default class ForgotPassword extends ForgotPasswordController {
    render() {
        return (
            // Required for all blocks
            <ForgotPassContainer>
                <Box className="logoContainerStyle" display="flex" flexDirection="row" justifyContent="space-between">
                    <Link to="/EmailAccountLogin">
                        <ArrowBack />
                    </Link>
                    <Link to="/LandingPage">
                        <img src={logoImg} alt="Logo" className="logoImgStyle" />
                    </Link>
                </Box>
                <Box className="contentForm">
                    <ForgotPassHeader
                        title="Forgot Password?"
                        paragraph="Enter your Registered email address to receive the OTP."
                    />
                    <Box display="flex" style={{ boxSizing: 'border-box' }}>
                        <Formik
                            initialValues={{
                                email: "",
                            }
                            }
                            validationSchema={this.forgotPassValidation}
                            onSubmit={this.sendEmailForForgotPass}

                        >
                            {this.forgotPassWordFormHandler}
                        </Formik>
                    </Box>
                </Box>
            </ForgotPassContainer>
        );
    }
}
