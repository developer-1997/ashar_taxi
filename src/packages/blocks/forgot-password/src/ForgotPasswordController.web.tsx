import React from "react";
import { Message } from "framework/src/Message";
import { BlockComponent } from "framework/src/BlockComponent";
import { runEngine } from "framework/src/RunEngine";
import MessageEnum, {
  getName
} from "framework/src/Messages/MessageEnum";

// Customizable Area Start
import { toast } from 'react-toastify';
import * as Yup from 'yup';
import { NewPasswordForm } from "../../../components/src/NewPasswrodForm.web";
import { ForgotPassForm } from "../../../components/src/ForgotPassForm.web";
import { OtpInputField } from "../../../components/src/OtpInputFieldForgotPass.web";
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  history: any;
  location: any;
  // Customizable Area End
}

export interface S {
  // Customizable Area Start
  forgotPassEmail: string,
  error: Object,
  email: string,
  disableBtn: boolean,
  disableResendBtn: boolean
  // Customizable Area End
}

export interface SS {
  // Customizable Area Start
  id: any;
  // Customizable Area End
}

export default class ForgotPasswordController extends BlockComponent<
  Props,
  S,
  SS
> {
  // Customizable Area Start
  forgotPassApiCallId: string = "";
  forgotPassOTPApiCallId: string = "";
  newpasswordApiCallId: string = "";
  newPassvalidation: {} = {};
  forgotPassValidation: {} = {};
  // Customizable Area End

  constructor(props: Props) {
    super(props);
    this.subScribedMessages = [
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.NavigationPayLoadMessage),
      getName(MessageEnum.CountryCodeMessage)
    ];
    this.receive = this.receive.bind(this);

    runEngine.attachBuildingBlock(this, this.subScribedMessages);

    this.state = {
      // Customizable Area Start
      forgotPassEmail: "",
      email: "",
      error: {},
      disableBtn: false,
      disableResendBtn: false,
      // Customizable Area End
    };

    // Customizable Area Start
    this.sendEmailForForgotPass = this.sendEmailForForgotPass.bind(this);
    this.createdNewPasswordFormHandler = this.createdNewPasswordFormHandler.bind(this);
    this.forgotPassWordFormHandler = this.forgotPassWordFormHandler.bind(this);
    this.forgotPassWordOtpHanlder = this.forgotPassWordOtpHanlder.bind(this);
    this.forgotPassValidation = Yup.object().shape({
      email: Yup.string().email('Please enter a valid email address').required('Required'),
    })

    this.newPassvalidation = Yup.object().shape({
      password: Yup.string().required('Required'),
      confirmpassword: Yup.string().required('Required').when("password", {
        is: val => (val && val.length > 0 ? true : false),
        then: Yup.string().oneOf(
          [Yup.ref("password")],
          "Password & Confirm Password does not match"
        )
      }),
    })

    // Customizable Area End
  }

  // Customizable Area Start
  async receive(from: string, message: Message) {
    if (getName(MessageEnum.RestAPIResponceMessage) === message.id) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );
      const sucessJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );
      const errorReponse = message.getData(
        getName(MessageEnum.RestAPIResponceErrorMessage)
      );

      if (sucessJson && !sucessJson.errors) {

        this.apiSuccessResponse(apiRequestCallId, sucessJson)

      } else if (sucessJson && sucessJson.errors) {

        this.apiErrorResponse(apiRequestCallId, errorReponse)

      }
    }
  }
  // Customizable Area End

  // Customizable Area Start

  forgotPassWordOtpHanlder({
    values,
    handleChange,
    handleSubmit,
  }:any){ 
    return (
    <OtpInputField
      values={values}
      handleChange={handleChange}
      handleSubmit={handleSubmit}
    />
  )}
  forgotPassWordFormHandler ({
    values,
    errors,
    touched,
    handleChange,
    handleSubmit,
    setFieldValue,
}:any) {
  return (
    <ForgotPassForm
        data-test-id="forgotPassWrapperId"
        values={values}
        handleChange={handleChange}
        handleFromValidation={this.handleFromValidation}
        errors={errors}
        touched={touched}
        state={this.state}
        setFieldValue={setFieldValue}
        handleSubmit={handleSubmit}
        disableBtn={this.state.disableBtn}
    />
)
}
  createdNewPasswordFormHandler ({
    values,
    errors,
    touched,
    handleChange,
    handleSubmit,
    setFieldValue,
}:any) {
  return (
      <NewPasswordForm
          values={values}
          handleChange={handleChange}
          handleFromValidation={this.handleFromValidation}
          errors={errors}
          touched={touched}
          state={this.state}
          setFieldValue={setFieldValue}
          handleSubmit={handleSubmit}
      />
  )
  }
  
  apiSuccessResponse = (apiCallId: any, sucessResponse: any) => {
    if (apiCallId === this.forgotPassApiCallId) {
      this.props.history.push(`/ForgotPasswordOTP`, { email: sucessResponse?.data.attributes.email, token: sucessResponse.meta.token })
      this.setState({
        disableBtn:false
      })
    }
    if (apiCallId === this.forgotPassOTPApiCallId) {
      this.props.history.push(`/NewPassword`, { email: this.state.forgotPassEmail })
    }
    if (apiCallId === this.newpasswordApiCallId) {
      this.props.history.push(`/EmailAccountLogin`)
    }
  }

  apiErrorResponse = (apiCallId: any, errorResponse: any) => {
    if (apiCallId === this.forgotPassApiCallId) {
      this.emailNotFound()
      this.setState({
        disableBtn:false
      })
    }
    if (apiCallId === this.forgotPassOTPApiCallId) {
      this.otpNotMatch()
    }
   
  }

  apiCall = async (data: any) => {
    const { contentType, method, endPoint, body, token } = data
    const header = {
      "Content-Type": contentType,
      token: token || ""
    }
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    )
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    )
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMessage),
      configJSON.baseURL
    )
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      method
    )
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      endPoint
    )
    body &&
      requestMessage.addData(
        getName(MessageEnum.RestAPIRequestBodyMessage),
        JSON.stringify(body)
      )
    runEngine.sendMessage(requestMessage.id, requestMessage)
    return requestMessage.messageId
  }

  sendEmailForForgotPass = async (attrs: any) => {
    this.setState({
      disableBtn:true
    })
    let payload = {}
    payload = {
      email: attrs.email,
    }
    this.forgotPassApiCallId = await this.apiCall({
      contentType: configJSON.forgotPasswordAPiContentType,
      method: configJSON.httpPostMethod,
      endPoint: configJSON.forgotPassEndPoint,
      body: payload
    })
  }
  newPassForForgotPass = async (attrs: any, email: any) => {
    let payload = {}
    payload = {
      password: attrs.password,
      password_confirmation: attrs.confirmpassword,
      email: email
    }
    this.newpasswordApiCallId = await this.apiCall({
      contentType: configJSON.forgotPasswordAPiContentType,
      method: configJSON.httpPostMethod,
      endPoint: configJSON.newPasswordEndPoint,
      body: payload
    })
  }


  handleOtpFormSubmit = (values: any, url: any, token:any) => {
    this.setState({
      forgotPassEmail: url,
    })
    let str = values.otp1 + values.otp2 + values.otp3 + values.otp4
    this.OtpVerificationForWeb(str, token)
  }

  OtpVerificationForWeb = async (attrs: any, token:any) => {
    console.log("token funct", token)
    this.forgotPassOTPApiCallId = await this.apiCall({
      contentType: configJSON.forgotPasswordAPiContentType,
      method: configJSON.httpPostMethod,
      endPoint: configJSON.forgotPassOTPEndPoint + "?pin=" + attrs,
      token: token
    })
  }

  emailNotFound = () => {
    toast.error("Email not found.")
  }
  otpNotMatch = () => {
    toast.error("Incorrect OTP")
  }
  otpSendAgain = () => {
    
    toast.info("OTP sent again")
    this.setState({
      disableResendBtn:true
    })
   
    setTimeout(() => {
      this.setState( {
        disableResendBtn: false
      })
    }, 120000);
    
  }
  handleFromValidation(error: any, touched: any) {
    return (
        <div className="errorText">
            {error && touched ? error : ""}
        </div>
    )
}
  // Customizable Area End
}
