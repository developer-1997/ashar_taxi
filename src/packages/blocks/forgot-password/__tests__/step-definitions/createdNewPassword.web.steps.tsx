//@ts-nocheck
import { defineFeature, loadFeature } from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'
import * as helpers from "../../../../framework/src/Helpers";
import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";
import MessageEnum, {
  getName
} from "../../../../framework/src/Messages/MessageEnum";

import React from "react";
import ForgotPasswordOTPconfirmation from "../../src/CreatedNewPassword.web"
import {NewPasswordForm} from "../../../../components/src/NewPasswrodForm.web"

const navigation = require("react-navigation")
jest.mock("formik")

const screenProps = {
    id: "newPassword",
    navigation: navigation,
    onchange: () => {},
    history: { push: jest.fn(), goBack: jest.fn() },
    location: {},
}


const newPasssuccessForm = {
    message: "Thank You, From submited has been successfully",
};
const newPassErrorForm = {
    errors: "Sorry, Form submited failed",
};

const feature = loadFeature('./__tests__/features/createdNewPassword-scenerio.feature');

defineFeature(feature, (test) => {
    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' } }))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
    });

    test('New password', ({ given, when, then }) => {
        let newPasswordWrapper : ShallowWrapper;
        let newPasswordFormWrapper: ShallowWrapper;
        let instance: ForgotPasswordOTPconfirmation;

        given('I am a User loading New password page', () => {
            newPasswordWrapper = shallow(<ForgotPasswordOTPconfirmation {...screenProps} />)
            expect(newPasswordWrapper).toBeTruthy()
            // expect(newPasswordFormWrapper).toBeTruthy()
        });

        when('I navigate to the New password Screen', () => {
            instance = newPasswordWrapper.instance() as ForgotPasswordOTPconfirmation
        });

        then('NewPassword page will load with out errors', () => {
            expect(newPasswordWrapper).toBeTruthy();
            expect(newPasswordWrapper).toMatchSnapshot();
        });
        then("I can handle the inputs in newPassword", () => {


            const newPassProps = {
                values: {
                    password: '',
                    confirmpassword:''
                },
                handleChange: jest.fn(),
                handleFromValidation: instance.handleFromValidation,
                errors: "",
                touched: "",
                state: {
                    enableReTypePasswordField: false
                },
                setFieldValue: jest.fn(),
                handleSubmit: instance.newPassForForgotPass,
            }
            newPasswordFormWrapper = shallow(<NewPasswordForm {...newPassProps} />)

            let PasswordInput = newPasswordFormWrapper.findWhere((node) => node.prop('data-test-id') === 'newFormPassword');
            PasswordInput.simulate('change', 'password')

            let ConfirmPasswordInput = newPasswordFormWrapper.findWhere((node) => node.prop('data-test-id') === 'newFormRetypePassword');
            ConfirmPasswordInput.simulate('change', 'passwordConfirm')
            
        })

        then("I can handle the submit button in newPassword",()=>{
            let verifyButton = newPasswordFormWrapper.findWhere((node) => node.prop('data-test-id') === 'newPassSubmitButton');
            verifyButton.simulate('click');
           
        })
        then("Check new password api is working",() => {
            const PostNewPasswordFormApi = new Message(
                getName(MessageEnum.RestAPIResponceMessage)
            );
            PostNewPasswordFormApi.addData(
                getName(MessageEnum.RestAPIResponceDataMessage),
                PostNewPasswordFormApi.messageId
            );
            PostNewPasswordFormApi.addData(
                getName(MessageEnum.RestAPIResponceSuccessMessage),
                JSON.parse(JSON.stringify(newPasssuccessForm))
            );
            PostNewPasswordFormApi.addData(
                getName(MessageEnum.RestAPIResponceSuccessMessage),
                JSON.parse(JSON.stringify({ status: 500 }))
            );
            instance.newpasswordApiCallId =
                PostNewPasswordFormApi.messageId;
            runEngine.sendMessage("Unit Test", PostNewPasswordFormApi);
            // ----
            PostNewPasswordFormApi.addData(
                getName(MessageEnum.RestAPIResponceSuccessMessage),
                JSON.parse(JSON.stringify({ errors: newPassErrorForm, }))
            );
            instance.newpasswordApiCallId =
                PostNewPasswordFormApi.messageId;
            runEngine.sendMessage("Unit Test", PostNewPasswordFormApi);
            // ------

            instance.newpasswordApiCallId =
                PostNewPasswordFormApi.messageId;
            runEngine.sendMessage("Unit Test", PostNewPasswordFormApi);
            instance.apiCall({
                contentType: "application/json",
                method: "",
                endPoint: "",
                body: "",
            });
        })
        then('I can leave the screen with out errors', () => {
            instance.componentWillUnmount();
            expect(newPasswordWrapper).toBeTruthy();
        });
    });
})
