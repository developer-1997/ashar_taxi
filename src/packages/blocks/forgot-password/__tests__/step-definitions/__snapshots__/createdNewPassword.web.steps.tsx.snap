// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newPassword New password 1`] = `
<ForgotPassContainer>
  <Styled(MuiBox)
    className="logoContainerStyle"
    display="flex"
    flexDirection="row"
    justifyContent="space-between"
  >
    <Link
      replace={false}
      to="/ForgotPassword"
    >
      <Memo(ForwardRef(ArrowBackIcon)) />
    </Link>
    <Link
      replace={false}
      to="/LandingPage"
    >
      <img
        alt="Logo"
        className="logoImgStyle"
        src={
          Object {
            "testUri": "../../../packages/blocks/email-account-registration/assets/logo.png",
          }
        }
      />
    </Link>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    className="verificationForm"
  >
    <ForgotPassHeader
      paragraph="Please set your new password, making sure it contains at least one uppercase letter, one lowercase letter, one number, and one special character for security"
      title="Reset password"
    />
    <Styled(MuiBox)
      display="flex"
      id="newPasswordForm"
    >
      <Formik
        initialValues={
          Object {
            "confirmpassword": "",
            "password": "",
          }
        }
        onSubmit={[Function]}
        validationSchema={
          ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_nodes": Array [
              "password",
              "confirmpassword",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "confirmpassword": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [
                  Condition {
                    "fn": [Function],
                    "refs": Array [
                      Reference {
                        "getter": [Function],
                        "isContext": false,
                        "isSibling": true,
                        "isValue": false,
                        "key": "password",
                        "map": undefined,
                        "path": "password",
                      },
                    ],
                  },
                ],
                "_deps": Array [
                  "password",
                ],
                "_exclusive": Object {
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
                "type": "string",
              },
              "password": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
                "type": "string",
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
            "type": "object",
          }
        }
      >
        <Component />
      </Formik>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</ForgotPassContainer>
`;
